src/cocotb/scheduler.py

```diff
@@ -809,7 +798,10 @@ def _schedule(self, task, trigger=None):
             trigger (cocotb.triggers.Trigger): The trigger that caused this
                 task to be scheduled.
         """
-        with self._task_context(task):
+        if self._current_task is not None:  # pragma: no cover
```

Comment:
I wouldn't exclude this line from coverage, it's an edge case that's clearly not tested (and could be, if we wanted to reach coverage closure). [Just to be clear: I don't think a test is needed as part of this PR.]